python/lsst/meas/base/diaCalculationPlugins.py

```diff
@@ -218,7 +218,8 @@ def calculate(self, diaObjects, diaSources, **kwargs):
         **kwargs
             Any additional keyword arguments that may be passed to the plugin.
         """
-        diaObjects.loc[:, "nDiaSources"] = diaSources.diaObjectId.count()
+        dtype = diaObjects["nDiaSources"].dtype
+        diaObjects.loc[:, "nDiaSources"] = diaSources.diaObjectId.count().astype(dtype)
```

Comment:
Is this coercion (and the ones below) necessary when you're going to apply `convertTableToSdmSchema` downstream anyway?